### PR TITLE
Remove libcue dependency

### PR DESCRIPTION
**Pull request by mahkoh**
Not ready but can be tested.

### CODE REVIEW

**Comment by Wang-Yue** [comment]
does this handle index 00 case?

**Comment by mahkoh** [comment]
It handles all cases in which index0 or index1. If one of them is missing and pregap is defined then the other one is calculated. It also adjusts for postgap. The results are somewhat different from what libcue produces. Maybe libcue ignores postgap. I have not looked into this.

**Comment by Wang-Yue** [comment]
LGTM

**Comment by Wang-Yue** [comment]
I tested your PR with many of my files and all seem fine by now.

**Comment by mahkoh** [comment]
Call, thanks. This won't go in before the next stable release, though. Unless @flyingmutant disagrees.

**Comment by flyingmutant** [comment]
I think it is OK to include it in the next release. It is a relatively localised change, which is unlikely to break much. And we'll probably catch the majority  of the errors and corner cases during -rc.
Do you consider it ready for review/merge?

**Comment by mahkoh** [comment]
No, I still have some local changes and need to split the commits.

**Comment by mahkoh** [comment]
Should be ready.

**Comment by mahkoh** [comment]
It should also be confirmed that this is in line with #251.

File: cue.c [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+	for (size_t i = 0; i < a_len; i++) {
+		if ((a[i] & ~0x20) != (b[i] & ~0x20))
+			return false;
+	}
+	return true;
+}
+
+static void cue_skip_spaces(struct cue_parser *p)
+{
+	while (p->len > 0 && (*p->src == ' ' || *p->src == '\t'))
```
Comment (flyingmutant) [inline cue.c]: Why not `isspace`?

File: cue.c [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+{
+	p->err = true;
+}
+
+static bool cue_str_eq(const char *a, size_t a_len, const char *b, size_t b_len)
+{
+	if (a_len != b_len)
+		return false;
+	for (size_t i = 0; i < a_len; i++) {
+		if ((a[i] & ~0x20) != (b[i] & ~0x20))
```
Comment (flyingmutant) [inline cue.c]: Comment or named constant for `~0x20` would make the reading a bit easier.

File: cue.c [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+
+	if (p->len > 0) {
+		char c = *p->src;
+		cue_consume(p);
+		if (p->len > 0 && c == '\r' && *p->src == '\n')
+			cue_consume(p);
+	}
+}
+
+static char *cue_strdup(const char *start, size_t len)
```
Comment (flyingmutant) [inline cue.c]: Why?

File: cue.c (hunk 1 above)
Comment (mahkoh) [inline cue.c]: Because isspace matches \n.

File: cue.c (hunk 3 above)
Comment (mahkoh) [inline cue.c]: Why not?

File: cue.c (hunk 2 above)
Comment (mahkoh) [inline cue.c]: Updated.

**Comment by Wang-Yue** [comment]
you also need to modify .travis.yml and https://github.com/cmus/cmus.github.io/blob/4d1d49fae9bf01fd6a9950a172b5e610e17916be/index.html

**Comment by Wang-Yue** [comment]
lgtm

**Pull request merged**